include/deal.II/lac/petsc_precondition.h

```diff
@@ -1090,6 +1080,67 @@ namespace PETScWrappers
     initialize();
   };
 
+  class PreconditionShell : public PreconditionBase
+  {
+  public:
+    /**
+     * Empty Constructor. You need to call initialize() before using this
+     * object.
+     */
+    PreconditionShell() = default;
+
+    /**
+     * Constructor. Take the matrix which is used to form the preconditioner.
+     */
+    PreconditionShell(const MatrixBase &matrix);
+
+    /**
+     * Same as above but without setting a matrix to form the preconditioner.
+     */
+    PreconditionShell(const MPI_Comm &communicator);
+
+
+    /**
+     * The callback for the application of the preconditioner. Defaults to
+     * a copy operation if not provided.
+     */
+    std::function<int(VectorBase &dst, const VectorBase &src)> apply;
+
+    /**
+     * The callback for the application of the transposed preconditioner.
+     * Defaults to the non-transpose operation if not provided.
+     */
+    std::function<int(VectorBase &dst, const VectorBase &src)> applyT;
```

Comment:
Would you mind renaming these functions to `vmult` and `Tvmult`? That is the notation used throughout deal.II.